src/EluvioLive.js

```diff
@@ -3045,11 +3045,29 @@ class EluvioLive {
       toJson = false;
     }
 
-    let res = await this.GetServiceRequest({
-      path: urljoin("/tnt/purchases/", tenant, marketplace),
-      queryParams: { offset, processor },
-      headers,
-    });
+    let res;
+    try {
+      res = await this.GetServiceRequest({
+        path: urljoin("/tnt/purchases/", tenant, marketplace),
+        queryParams: { offset, processor },
+        headers,
+      });
+    } catch (e) {
+      console.error(e);
+      try {
+        // if unauthorized try again with inactive marketplace flag
+        if (e.status == 401) {
+          res = await this.GetServiceRequest({
```

Comment:
we need to leave the old res in place, regular users shouldn't realize a second query took place

Reply:
there is no old res if we get the first catch block.

i changed to logging to not console.log(e) or (err2)

```
      } catch (err2) {
        console.log("Error: unauthorized or invalid tenant or marketplace");
        // throw original error
        throw e;
      }

```